omnisafe/envs/core.py

```diff
@@ -217,16 +218,25 @@ def __getattr__(self, name: str) -> Any:
     def step(
         self, action: torch.Tensor
     ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, Dict]:
-        return self._env.step(action)
+        obs, reward, cost, terminated, truncated, info = self._env.step(action)
+        return (
```

Comment:
change device only in `safety_gymnasium_env.py` please, changing device in `Wrapper` could cause multi-calculating in `reset` and `step`